generic_expression_patterns_modules/lv.py

```diff
@@ -0,0 +1,546 @@
+    """
+    This function returns the dictionary of generic genes and specific genes
+    that were included in the multiPLIER or eADAGE analyses. We want to make
+    sure that our gene lists obtained from SOPHIE vs multiPLIER or eADAGE
+    are consistent. This will prevent indexing by a gene that doesn't
+    exist and resulting in NA values downstream.
+
+    Arguments
+    ---------
+    dict_genes: dict
+        Dictionary mapping gene ids to label="generic", "other"
+
+    LV_matrix: df
+        Dataframe containing contribution of gene to LV (gene x LV matrix)
+    """
+    model_genes = list(LV_matrix.index)
+
+    processed_dict_genes = {}
+    for gene_label, ls_genes in dict_genes.items():
+        ls_genes_processed = list(set(model_genes).intersection(ls_genes))
+
+        processed_dict_genes[gene_label] = ls_genes_processed
+
+    return processed_dict_genes
+
+
+def get_nonzero_LV_coverage(dict_genes, LV_matrix):
+    """
+    This function counts the number of LVs that each
+    gene is present in (i.e. has a nonzero contribution).
+    This function returns a dictionary [gene id]: number of LVs
+
+    Arguments
+    ---------
+    dict_genes: dict
+        Dictionary mapping gene ids to label="generic", "other"
+
+    LV_matrix: df
+        Dataframe containing contribution of gene to LV (gene x LV matrix)
+    """
+    dict_nonzero_coverage = {}
+    for gene_label, ls_genes in dict_genes.items():
+        LV_series = (LV_matrix.loc[ls_genes] != 0).sum(axis=1)
+
+        dict_nonzero_coverage[gene_label] = LV_series
+
+    return dict_nonzero_coverage
+
+
+def get_highweight_LV_coverage(dict_genes, LV_matrix, quantile=0.9):
+    """
+    This function count the number of LVs that each
+    gene contributes a lot to (i.e. has a high negative or positive
+    weight contribution).
+    This function returns a dictionary [gene id]: number of LVs
+
+    Note: Using the quantile means that each LV has the same number
+    of high weight values. Also here we are using a quantile cutoff
+    since our distribution is not normal (exponential PDF)
+
+    Arguments
+    ---------
+    dict_genes: dict
+        Dictionary mapping gene ids to label="generic", "other"
+
+    LV_matrix: df
+        Dataframe containing contribution of gene to LV (gene x LV matrix)
+
+    quantile: float(0,1)
+        Quantile to use to threshold weights. Default set to 90th quantile.
+    """
+    thresholds_per_LV = LV_matrix.quantile(quantile)
+
+    # Manually checked that genes selected as high weight
+    # are above threshold using below print statements
+    # print(thresholds_per_LV)
+    # print(LV_matrix)
+    # print(
+    #    LV_matrix.loc[
+    #        (LV_matrix.abs() > thresholds_per_LV)["Node2"].values, "Node2"
+    #    ]
+    # )
+    dict_highweight_coverage = {}
+    for gene_label, ls_genes in dict_genes.items():
+
+        LV_series = (LV_matrix.abs() > thresholds_per_LV).sum(axis=1)[ls_genes]
```

Comment:
The abs is actually not necessary in this PLIER case because the distribution of weights are all nonzero. But in the future if someone wanted to use this analysis, Casey pointed out that when I was looking at nonzero weights I forgot to include negative weights

![image](https://user-images.githubusercontent.com/13107360/105760087-415a6480-5f1f-11eb-9c2b-df8c91dca758.png)